feat: indicate tale import status via the Run button

bodom0015 opened the pull request:
## Problem
Run View contains stale data after Tale import completes - needs manual refresh to pick up newly-registered datasets, related identifiers, etc

Fixes #137 
Fixes #121 

## Approach
* Wire up the Run Tale view to the SyncService to refresh Tale data when import completes for the Tale

## How to Test
1. Checkout and run this branch locally, rebuild the dashboard
2. Login to the WholeTale Dashboard
3. Import a Tale (preferably a large Tale that take awhile to import and/or ultimately fails)
    * You should be brought to the Run view
    * You should see that the "Run Tale" button is disabled and reads `Importing...`
4. Wait for the Tale import to finish
    * On failure, you should see that the button stays disabled and reads `Import Failed`
    * On success, you should see the "Run Tale" button now appears as expected, allowing the user to interact with it to launch the Tale

craig-willis commented:
I think the stale "datasets used" and "related identifiers" were already handled in an earlier PR with the `wt_tale_updated` implementation, although I see no harm in specifically handling the completed event. For me what remains of https://github.com/whole-tale/ngx-dashboard/issues/137 is some indication that the tale import is incomplete.

bodom0015 commented:
Ah, sorry.. I think I had misread your comment here and thought there was an edge case here that we weren't handling: https://github.com/whole-tale/ngx-dashboard/issues/137#issuecomment-799668756

bodom0015 commented:
Updated this PR to show the Tale Import status on the Run view, and updated test case above.

After successful import, you should see the "Run Tale" button appear as it does currently

During import, you should see a disabled button with a spinner that reads `Importing...`:
<img width="370" alt="Screen Shot 2021-05-25 at 3 37 42 PM" src="https://user-images.githubusercontent.com/1413653/119565251-5d580080-bd6f-11eb-8470-aa64c9bb8386.png">

On a failed import, you should see the button is still disabled (open question: how does the user resolve a failed import?):
<img width="329" alt="Screen Shot 2021-05-25 at 3 38 07 PM" src="https://user-images.githubusercontent.com/1413653/119565474-9a23f780-bd6f-11eb-8900-59fbee8a4417.png">

craig-willis reviewed (approved):
I tested using https://dashboard.local.wholetale.org/browse?uri=https%3A%2F%2Fdoi.org%2F10.5281%2Fzenodo.4774875&name=Test and selecting READ/WRITE, the copy workspace takes a little time.  I previously used https://doi.org/10.5281/zenodo.4782234, which is very large and caused an error during import.  I see the button state changes as expected.  

> how is the user expected resolve a failed import? 

My expectation is that they delete and reimport (for now)

My initial reaction was that "Importing..." and "Import failed" are too specific, given other processes may change the tale status.  The only other call that I see is `copyTale`, but it doesn't have notifications, so maybe not worrying about for now.

Part of me thinks "Importing..." and "Import failed" are too specific. Looking at girder_wholetale, most of the tale status changes are from import tasks, but also copyTale. But copy doesn't have notifications

bodom0015 commented:
The labels on the Girder side are `PREPARING` instead of "Importing...", and `ERROR` instead of "Import Failed"

I am unclear on when else these labels would be applied, outside of the Import. For example, currently if the build fails the user can still click the Run button to retry the build - the same is not true with Import, which appears to be the outlying special case here?

I'm certainly open to changing the wording there if there are cases outside of import when this would be applicable

bodom0015 commented on src/app/tales/components/tale-run-button/tale-run-button.component.html:
```diff
@@ -1,22 +1,32 @@
-</div>
\ No newline at end of file
+<div>
+  <button *ngIf="tale && tale.status === 0" class="right floated" [ngClass]="{ 'tale-button disabled': !isPrimary, 'tale-button-primary disabled': isPrimary }" disabled="true">
+      <i class="fas fa-spinner fa-pulse"></i>
+      Importing...
+  </button>
+  <button *ngIf="tale && tale.status === 2" class="right floated" [ngClass]="{ 'tale-button disabled': !isPrimary, 'tale-button-primary disabled': isPrimary }" disabled="true">
+      <i class="warning icon"></i>
+      Import Failed
+  </button>
```
```suggestion
  <button *ngIf="tale && tale.status === 0" class="right floated" [ngClass]="{ 'tale-button disabled': !isPrimary, 'tale-button-primary disabled': isPrimary }" disabled="true">
      <i class="fas fa-spinner fa-pulse"></i>
      Preparing...
  </button>
  <button *ngIf="tale && tale.status === 2" class="right floated" [ngClass]="{ 'tale-button disabled': !isPrimary, 'tale-button-primary disabled': isPrimary }" disabled="true">
      <i class="warning icon"></i>
      Initialization Failed
  </button>

```
Note that the "Error" label is already applied in some cases where the instance fails to launch. Such cases are correctable by deleting and relaunching (e.g. pressing the Error button and then clicking Run Tale again).

To distinguish such events from failed imports / failed copies, I'd suggest using something slightly more explicit here
